fuel-txpool/src/containers/dependency.rs

```diff
@@ -392,17 +392,16 @@ impl Dependency {
 
                             Self::check_if_coin_input_can_spend_db_coin(&coin, input)?;
                         }
-
                         max_depth = core::cmp::max(1, max_depth);
-                        db_coins.insert(
-                            *utxo_id,
-                            CoinState {
-                                is_spend_by: Some(tx.id() as TxId),
-                                depth: 0,
-                            },
-                        );
                     }
-
+                    // mark this coin as spent by the current tx
+                    db_coins.insert(
```

Comment:
You will overwrite state, also in the case if `state.is_spend_by` is `None`. 

You only need to do that if you really have a conflict

Reply:
shouldn't we still mark the coin as spent by this tx even if it was previously none?

For example, if I have a tx2 that depends on a tx1 output, that output should now be marked as spent by tx2 even without any conflicts.

Reply:
I already dismissed my review=)